src/components/TaskComponent.js

```diff
@@ -0,0 +1,19 @@
+import React from "react";
+
+const TaskComponent = ({ task, deleteTask, editTask}) => {
+  const handleEdit = (task) => {
+    editTask(task);
```

Comment:
This should be renamed to `showEditModal`. It is not editing the task but just setting state and opening the modal in edit state.